### PR TITLE
Mono: Use PascalCase in core types.

### PR DESCRIPTION
**Pull request by 27thLiz**
Supersedes #12736

### CODE REVIEW

File: modules/mono/glue/cs_files/Color.cs [hunk 1]
```diff
@@ -71,36 +71,36 @@ public float h
             }
             set
             {
-                this = from_hsv(value, s, v);
+                this = FromHsv(value, s, v);
```
Comment (akien-mga) [inline modules/mono/glue/cs_files/Color.cs]: Shouldn't it be `FromHSV`?

File: modules/mono/glue/cs_files/Color.cs [hunk 2]
```diff
@@ -154,10 +154,10 @@ public Color Black
             }
         }
 
-        public static void to_hsv(Color color, out float hue, out float saturation, out float value)
+        public static void ToHsv(Color color, out float hue, out float saturation, out float value)
```
Comment (akien-mga) [inline modules/mono/glue/cs_files/Color.cs]: And here `ToHSV`.

File: modules/mono/glue/cs_files/Color.cs [hunk 3]
```diff
@@ -290,7 +290,7 @@ public int to_32()
             return c;
         }
 
-        public int to_ARGB32()
+        public int ToArgb32()
```
Comment (akien-mga) [inline modules/mono/glue/cs_files/Color.cs]: Here `ToARGB32`, below `ToHTML`.. Or what would be the convention in such cases?

Comment (Hinsbart) [reply modules/mono/glue/cs_files/Color.cs]: Yeah, I wasn't sure about these either.
The way it is now is using the default visual studio/resharper naming convention.

Comment (akien-mga) [reply modules/mono/glue/cs_files/Color.cs]: It looks like the style you used is the recommended one: https://msdn.microsoft.com/en-us/library/ms229043%28v=vs.100%29.aspx#Capitalization%20Rules%20for%20Acronyms

**Review by akien-mga — APPROVED** [review]

**Review by neikeq — APPROVED** [review]

**Pull request merged**